hardware_enablement/kmm-release-notes.adoc

```diff
@@ -28,4 +27,297 @@ toc::[]
+
+** *Consequence*: Tolerations other than Kernel Module Management-specific tolerations can appear in the status.
+
+** *Fix*: The NMC status now gets its toleration from a dedicated annotation rather than from the worker pod.
+
+** *Result:* The NMC status only contains the Module's tolerations.
+
+// MGMT-20725
+* The KMM Operator version 2.4 fails to start properly and cannot list the resource `\modulebuildsignconfigs\`. 
+
+** *Cause*: On the Kernel Module Management (KMM) Operator, when the Operator is installed using Red Hat Konflux it does not start properly as the log files contain errors.
+
+** *Consequence*: The KMM Operator does not work as expected.
+
+** *Fix*: The Cluster Service Version (CSV) file was updated to list the resources `\modulebuildsignconfigs\` and `moduleimagesconfig`.
+
+** *Result:* The KMM Operator works as expected.
+
+// MGMT-20752
+* Konflux build does not include version and git commit in the operator logs.
+
+** *Cause*: On the Kernel Module Management (KMM) Operator, when the Operator was built using Communications Platform as a Service (CPaas), the build included the Operator version and git commit in the log  files. However, with Red Hat Konflux these details are not included in the log files.
+
+** *Consequence*: Important information is missing from the log files.
+
+** *Fix*: Some adjustments were made in Konflux.
+
+** *Result:* The KMM Operator build now includes the Operator version and git commit in the log files.
+
+// MGMT-20754
+* The KMM Operator does not load the module after node with taint is rebooted.
+
+** *Cause*: The Kernel Module Management (KMM) Operator does not reload the kernel module in case the node reboot sequence is too quick. The reboot is determined based on the timestamp of the status condition being later than the timestamp in the Node Machine Configuration (NMC) status.
+
+** *Consequence*: When the reboot happens quickly, in less time than the grace period, the node state will not change. After the node reboots, KMM will not load the kernel module again.
+
+** *Fix*: Instead of relying on the condition state, NMC can rely on the `Status.NodeInfo.BootID` field. This field is set by kubelet based on the /proc/sys/kernel/random/boot_id file of the server node, so it is updated after each reboot.
+
+** *Result:* The more accurate timestamps enable the Kernel Module Management (KMM) Operator to reload the kernel module after the node reboot sequence.
+
+// MGMT-20775
+* Redeploying a module that uses in-cluster builds fails with the `ImagePullBackOff` policy.
+
+** *Cause*: On the Kernel Module Management (KMM) Operator, the `pullPolicy` for the puller pod and the worker pod is different.
+
+** *Consequence*: An image can be considered as existing while it fact it is not.
+
+** *Fix*: Make the pull policy of the pull pod the same as the pull policy defined in the KMM Module, as this is what is used by the worker pod.
```

Comment:
```suggestion
** *Fix*: Make the image pull policy of the pull pod the same as the pull policy defined in the KMM module since its the same policy that is used by the worker pod.
```

Reply:
changed